pkg/artifactory/resource/replication/resource_artifactory_pull_replication_test.go

```diff
@@ -40,22 +42,70 @@ func mkTclForPullRepConfg(name, cron, url string) string {
 
 func TestAccPullReplicationInvalidCron(t *testing.T) {
 
-	_, fqrn, name := test.MkNames("lib-local", "artifactory_pull_replication")
+	_, _, name := test.MkNames("lib-local", "artifactory_pull_replication")
 	var failCron = mkTclForPullRepConfg(name, "0 0 * * * !!", acctest.GetArtifactoryUrl(t))
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acctest.PreCheck(t) },
 		ProviderFactories: acctest.ProviderFactories,
-		CheckDestroy:      testAccCheckReplicationDestroy(fqrn),
 		Steps: []resource.TestStep{
 			{
 				Config:      failCron,
-				ExpectError: regexp.MustCompile(`.*syntax error in year field: '!!'.*`),
+				ExpectError: regexp.MustCompile(`.*Invalid cronExp.*`),
 			},
 		},
 	})
 }
 
+func TestAccCronExpressions(t *testing.T) {
+	cronExpressions := [...]string{
+		"10/20 15 14 5-10 * ? *",
+		"* 5,7,9 14-16 * * ? *",
+		"* 5,7,9 14/2 ? * WED,Sat *",
+		"* * * * * ? *",
+		"* * 14/2 ? * mon/3 *",
+		"* 5-9 14/2 ? * 1-3 *",
+		"*/3 */51 */12 */2 */4 ? *",
+		"* 5 22-23 ? * Sun *",
+		"0/5 14,18,3-39,52 * ? JAN,MAR,SEP MON-FRI 2002-2010",
+		"0 15 10 * * ? *",
+		"0 15 10 ? * 6#2",
+		"0 15 10 15 * ?",
+		"0 0 2 ? * MON-SAT",
+	}
+	for _, cron := range cronExpressions {
+		title := fmt.Sprintf("TestPullReplicationLocalRepoCron_%s", cases.Title(language.AmericanEnglish).String(cron))
+		t.Run(title, func(t *testing.T) {
+			resource.Test(PullReplicationLocalRepoTestCase(cron, t))
+		})
+	}
+}
+
+func PullReplicationLocalRepoTestCase(cronExpression string, t *testing.T) (*testing.T, resource.TestCase) {
```

Comment:
Let's switch to lowercase func name as we aren't exporting this func.

Reply:
agree, done.